internal/provider/cluster_resource.go

```diff
@@ -736,6 +749,17 @@ func (r *clusterResource) Update(
 		clusterReq.SetParentId(parentID)
 	}
 
+	if !(plan.DeleteProtection.IsNull() || plan.DeleteProtection.IsUnknown()) &&
```

Comment:
did you find that these checks were still necessary after adding the default?

Reply:
Oh I forgot about that. Let me see if I can verify it.

Reply:
I thought about this some more and I think I actually do want this, and I also don't want a default value on the field because it causes the plan to always have a `false` value if the spec leaves the field unset. In that case I actually just want to let the server decide the state and for TF it leave it alone.